Feature/view order details brun32700

brun32700 opened the pull request:
Resolves #51

Summary:
-> Removed list of products of an order from the ViewOrders() method.
-> Added list of products of an order to the ViewOrderDetails() method using provided order Id.
-> After being shown list of products of an order, the user is then asked whether they want to view another orders, order details. If not, they are sent back to the main menu.
-> Added GetOrder(int id) in KebabController, to retrieve a single order by its Id.

Observations:
-> _userInput.GetValidAnswer() - there is no case sensitivity, or spacing (trim) check, is this intended? (so Yes, YES doesn't work etc.)
-> _userInput.GetValidAnswer() - When wrong value is inserted, the user is not informed that there is a wrong value (not user friendly)

Please review my pull request, and let me know if I missed something or something needs improvement.
Thanks in advance.

brun32700 committed:
Added GetOrder using id method

brun32700 committed:
ViewOrderDetails modified ask for id and display order details

brun32700 committed:
Removed order details from ViewOrders

brun32700 committed:
ViewOrderDetails completed displaying order details and asking to view another

brun32700 committed:
Improved spacing of console output text in ViewOrderDetails()

TheCSharpAcademy reviewed (changes requested):
@brun32700 At the moment we create a new index when showing orders so it doesn't reflect the Id of the order in the database (that is expected).

![image](https://user-images.githubusercontent.com/108711476/222011397-943d82a7-7b12-411d-bbf1-28ba0c67a9b4.png)

![image](https://user-images.githubusercontent.com/108711476/222011419-bf93ac02-aa6c-4710-9b7b-90a84ccaf9ac.png)

You'll need to make sure the "fake" id is translated into the correct Id when I choose an order. You can do that by pulling the orders list into the ViewOrderDetail method, which will allow you to choose the correct order. So if my choice is  'var mychoice = 1', you'll have to do GetOrderId(orderList[myChoice-1].Id). I subtract one since lists in c# are zero based. 

Let me know if you need help.

brun32700 committed:
Removed GetOrder(int id) as unused

brun32700 committed:
View order details by index from orders list

brun32700 commented:
@TheCSharpAcademy I have updated the method to use the index for getting the order details, if you could please re-check it.

So I pulled the orders list into the ViewOrderDetails() method as you suggested and call the ViewOrders() method to display the orders to the user. But now I am querying the orders list in both of the methods. 

Would it be better if we adjusted the ViewOrders() method to pass in an orders list to be displayed? Because the orders in ViewOrderDetails() and the orders in ViewOrders() could be different, as new orders are being added constantly (assuming multiple users are using this app simultaneously)?

I could just display the orders using the list in the ViewOrderDetails() method, but then I would be repeating the function of ViewOrders().

![image](https://user-images.githubusercontent.com/48837066/222077654-f633bcfc-e74f-4e14-92cd-32f12cb0a7a7.png)

TheCSharpAcademy commented:
> Would it be better if we adjusted the ViewOrders() method to pass in an orders list to be displayed? Because the orders in ViewOrderDetails() and the orders in ViewOrders() could be different, as new orders are being added constantly (assuming multiple users are using this app simultaneously)?
> 

@brun32700  Yeah that works! View orders eventually will live in a UI class, but right now you're right, it would be unnecessary code repetition.

brun32700 committed:
Adjusted ViewOrders method to pass in list of orders

brun32700 commented:
@TheCSharpAcademy I have changed the ViewOrders() method to accept a list of orders to print out to the console, instead of getting the list from within the method. Is this good now?

TheCSharpAcademy commented:
@brun32700 I'll approve the PR but we were still grabbing the wrong Id. Check out the modification: 

![image](https://user-images.githubusercontent.com/108711476/222274700-159fd43c-f943-4c31-88d2-32a983016fd2.png)

brun32700 commented:
@TheCSharpAcademy In my eyes it seems to be the correct Id (on my side anyway). See images below:

Unless you mean that I should not assume that the list of orders will be displayed in the same order (could be reversed in ViewOrdersMethod etc), so that is why we need to get the actual Id of the order and not the index of the order in the list?

If this is the case then, where should I submit a pull request to? Do I need to create a new issue for it?

![image](https://user-images.githubusercontent.com/48837066/222343420-2a940ee0-93c6-4b5b-9de9-f779caea04c4.png)
![image](https://user-images.githubusercontent.com/48837066/222343581-f22ee0bb-3c18-4d30-b436-795fe483255c.png)
